aviatrix/resource_aviatrix_kubernetes_cluster.go

```diff
@@ -0,0 +1,291 @@
+				Description:  "Region where the cluster is deployed.",
+				RequiredWith: []string{"account_id", "account_name", "name", "vpc_id", "version", "platform", "is_publicly_accessible", "network_mode"},
+			},
+			"version": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Version of the Kubernetes cluster.",
+				RequiredWith: []string{"account_id", "account_name", "name", "vpc_id", "region", "platform", "is_publicly_accessible", "network_mode"},
+			},
+			"platform": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Platform of the Kubernetes cluster, e.g. kops, kubeadm or any other free string.",
+				RequiredWith: []string{"account_id", "account_name", "name", "vpc_id", "region", "version", "is_publicly_accessible", "network_mode"},
+			},
+			"is_publicly_accessible": {
+				Type:         schema.TypeBool,
+				Optional:     true,
+				Description:  "Whether the API server is publicly accessible outside the virtual network.",
+				RequiredWith: []string{"account_id", "account_name", "name", "vpc_id", "region", "version", "platform", "network_mode"},
+			},
+			"network_mode": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Network mode of the cluster. Possible values are FLAT, OVERLAY.",
+				ValidateFunc: validation.StringInSlice([]string{"FLAT", "OVERLAY"}, false),
+				RequiredWith: []string{"account_id", "account_name", "name", "vpc_id", "region", "version", "platform", "is_publicly_accessible"},
+			},
+			"project": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Project name if the cluster is deployed in GCP.",
+			},
+			"compartment": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Compartment id if the cluster is deployed in OCI.",
+			},
+			"tags": {
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "Map of tags.",
+			},
+		},
+	}
+}
+
+// parseArn parses the ARN and returns the account ID, region and name
+//
+// Example:
+//
+//	parseArn("arn:aws:eks:us-east-2:123456789012:cluster/testcluster") => "123456789012", "us-east-2", "testcluster", nil
+func parseArn(arn string) (accountId string, region string, name string, err error) {
+	if !strings.HasPrefix(arn, "arn:") {
+		err = errors.New("ARN must start with 'arn:'")
+		return
+	}
+	sections := strings.SplitN(arn, ":", 6)
+	if len(sections) != 6 {
+		err = errors.New("ARN must have 6 sections")
+	}
+	return sections[4], sections[3], strings.TrimPrefix(sections[5], "cluster/"), nil
+}
+
+func marshalKubernetesClusterInput(d *schema.ResourceData) (*goaviatrix.KubernetesCluster, error) {
+	var clusterId string
+	arn, ok := d.GetOk("arn")
+	if ok {
+		accountId, region, name, err := parseArn(arn.(string))
+		if err != nil {
+			return nil, err
+		}
+		clusterId = fmt.Sprintf("%s-%s-%s", accountId, region, name)
```

Comment:
Why can't the 'arn' be used directly as an ID?

Reply:
Because CAI currently computes the resourceId like this.
Maybe we can also change CAI, but for how it is right now, the resourceId is computed like this.

Reply:
Robert, please resolve these remaining conversation and I'll merge.

Reply:
Resolving.
If we want to change it, we would need more changes in CAI.
And we could also change it at a later time.